### PR TITLE
Fix saving uuid as item title

### PR DESCRIPTION
**Pull request by PenghaiZhang**
This bug happens in some cases when hitting Save button the XML of an item is still not available .

### CODE REVIEW

**Comment by abidingotter** [comment]
Not sure this is the correct fix.  I mean, it may work, but I think the problem comes from the XML doc being out of date because the client script getState promise only runs once.
There is also still the problem of validation hooks not being unloaded, but that's a different issue :)

**Comment by PenghaiZhang** [comment]
> Not sure this is the correct fix. I mean, it may work, but I think the problem comes from the XML doc being out of date because the client script getState promise only runs once.
> There is also still the problem of validation hooks not being unloaded, but that's a different issue :)

Yes, getState promise only runs once. So I try to get it running again inside the "presubmit". And then I found the real reason is the item's XML is not really ready when calling getState.
If you go to line 106 of WizardApi, you would see 'itemPack.getXml ' actually does not have the XML we need yet.

**Comment by abidingotter** [comment]
Something I would test is:
1. Put a repeater on the first page and add some repeater items. 
2. Now remove a repeater item.  
3. Now save + publish. 
4. Check the item you removed is not in the XML

And:
1. Put some optional controls on page 2
2. Go to page 2 and fill in some values
3. Go back to page 1 
4. Go back to page 2 and put in some new values
5. Now save + publish. 
6. Check that the first set of values you put on page 2 are not in the XML.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/wizard/section/PagesSection.java [hunk 1]
```diff
@@ -121,7 +122,10 @@ public void handleWizardState(SectionInfo info, final WizardState state) {
     if (model.isSubmit()) {
       String xmlDoc = model.getXmldoc();
       if (xmlDoc != null) {
-        state.setItemXml(xmlDoc);
+        PropBagEx itemXml = state.getItemPack().getXml();
```
Comment (abidingotter) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/wizard/section/PagesSection.java]: state.getItemxml() should be sufficient

**Comment by PenghaiZhang** [comment]
> Something I would test is:
> 
> 1. Put a repeater on the first page and add some repeater items.
> 2. Now remove a repeater item.
> 3. Now save + publish.
> 4. Check the item you removed is not in the XML
> 
> And:
> 
> 1. Put some optional controls on page 2
> 2. Go to page 2 and fill in some values
> 3. Go back to page 1
> 4. Go back to page 2 and put in some new values
> 5. Now save + publish.
> 6. Check that the first set of values you put on page 2 are not in the XML.

Thanks Aaron. Let me test these two cases now.

**Comment by PenghaiZhang** [comment]
For the second test, looks good. 
Screenshot from 2019-07-23 10-05-24
Screenshot from 2019-07-23 10-05-36
Screenshot from 2019-07-23 10-05-59

**Comment by PenghaiZhang** [comment]
For the Repeater test : 
Screenshot from 2019-07-23 10-21-13
Screenshot from 2019-07-23 10-18-46
Screenshot from 2019-07-23 10-18-33

**Pull request merged**